### ISSUE TITLE
[COMMUNITY REVIEW] Deprecate metadata parameters in favor of json for metadata cli

### ISSUE THREAD

**Issue by lresende** [elyra-ai/elyra#2376]
**Is your feature request related to a problem? Please describe.**
We have been discussing improvements on the Metadata CLI and one of the options would be to start using files as the input for the CLI instead of using parameters for each field which is not so flexible and flat. Using files also would help in importing/exporting existing metadata configuration from the filesystem (e.g. moving to new envs, etc)

The idea would be to deprecate the current functionality on 3.x and remove it on 4.0

Please express your opinion in the comments.

**Comment by kevin-bates** [comment]
Thanks for opening this @lresende. 

I think the richness inherent in JSON schema introduces a sustainability issue with attempting to interpret, then convert, each field of a schema property to a CLI option.  Since we now allow folks to bring their own schemas, we have less control over what constructs are used in the schema.  As a result, I think, at a minimum, we need a means of allowing a direct bypass of the schema-to-options processing to occur.

This will, however, affect users.  

- No longer will there be a rich "usage statement" that lists the schema properties as parameters, including their description, type, and required-ness.
- Schema validation errors are cryptic and we'd be at the mercy of these for most things that are caught today prior to submission to the metadata service.
- Instance creation requires knowledge of the JSON syntax, where quoted strings, lists, object-valued properties will be difficult for users.

We need to make sure there's also a way to create instances directly (not using files), so I would propose both a `--file` and a `--json` option with the latter treating the CLI value as a JSON string (and using `json.loads()` to apply the JSON syntax interpretation).

Regarding import and export, I believe these should be separate sub-commands for the tooling where import can read from an export "sink" (most likely a directory).

One of the potential benefits of using a more "raw" approach, is that we could entertain the idea of using the REST API instead (I suppose we don't preclude doing this today, although it requires a running instance).  This would then allow for admins to more easily replicate configurations by not having to be local to each system.

For example, the following would replicate instances of `my_schema` from the system running at `foo` to that running at `bar`:
```
elyra-metadata export --schema_name=my_schema --endpoint=http://foo:port --directory=my_exported_data
elyra-metadata import --schema_name=my_schema --endpoint=http://bar:port --directory=my_exported_data
```

I guess I'm torn since MOST schemas we need to support will be fairly straightforward and placing the burden on the user feels unwarranted.  I'm wondering if we could _detect_ unsupported constructs such that attempts to use the schema-to-options approach will be intercepted with an exception that then suggests either `--file` or `--json`.

**Comment by lresende** [comment]
> Regarding import and export, I believe these should be separate sub-commands for the tooling where import can read from an export "sink" (most likely a directory).

Yeah, import/export is a parallel discussion and should not be impacted by this decision.
